sdk/core/azure-core/src/http/curl/curl_session_private.hpp

```diff
@@ -197,7 +197,7 @@ namespace Azure { namespace Core { namespace Http {
      * `STREAMING` and the response has been read completely.
      *
      */
-    SessionState m_sessionState;
+    SessionState m_sessionState = SessionState::PERFORM;
```

Comment:
I am surprised to see this as the default value. Should we add a third, `SessionState::None` or `SessionState::NotStarted`?

Reply:
There is not such state for a session. As soon as a session is created, it is on PERFORM state.
Right now there are just two relevant states for a session, whenever it is STREAMING or not.
I was going to use just  boolean, but I wanted to leave it ready just in case there is a new meaningful state that need to be added.